app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt

```diff
@@ -116,37 +126,204 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
+        showProgress(getString(R.string.scan_available_wifi))
+        deviceConnectPresenter.searchWiFi()
+    }
+
+    override fun showDialog(message: String, title: String) {
+        val dialogBuilder = context?.let { it1 -> AlertDialog.Builder(it1) }
+        dialogBuilder?.setMessage(message)
+                ?.setCancelable(false)
+                ?.setPositiveButton("OK", DialogInterface.OnClickListener { dialog, id ->
+                    dialog.cancel()
+                })
+        val alert = dialogBuilder?.create()
+        alert?.setTitle(title)
+        alert?.show()
+    }
+
+    // Function to show available rooms in the 3rd step
     override fun rooms() {
+        stopProgress()
         room.visibility = View.VISIBLE
         addDeviceButton.visibility = View.GONE
         deviceList.visibility = View.GONE
         wifiList.visibility = View.GONE
+        password_layout.visibility = View.GONE
+        showWifiList = false
+        connection_susiai_main_screen.visibility = View.GONE
+        showWifi.visibility = View.GONE
+        room_wizard.background = context?.let { ContextCompat.getDrawable(it, R.drawable.border_blue) }
+        account_wizard.background = context?.let { ContextCompat.getDrawable(it, R.drawable.border_normal) }
+        account_wizard.setTextColor(Color.BLACK)
+        room_wizard.setTextColor(Color.WHITE)
         showRooms()
 
+        if (roomNameSelected.isNullOrEmpty()) {
+            context?.let { ContextCompat.getColor(it, R.color.default_bg) }?.let { room_next.setBackgroundColor(it) }
+            room_next.setTextColor(Color.BLACK)
+            room_next.isClickable = false
+        }
+
         add_room.setOnClickListener {
             val roomName = edt_room.text.toString()
             if (!roomName.isEmpty()) {
                 deviceConnectPresenter.addRoom(roomName)
-                Toast.makeText(context, "Added " + roomName + " as a room", Toast.LENGTH_SHORT).show()
                 edt_room.text.clear()
                 Utils.hideSoftKeyboard(context, rootView)
             }
         }
+
+        room_next.setOnClickListener {
+            if (!roomNameSelected.isNullOrEmpty()) {
+                // To call the next screen when room name is selected
+                // Send the room name to the speaker by making another endpoint
+                deviceConnectPresenter.makeAddRoomRequest(roomNameSelected.toString())
+            }
+        }
+
+        room_previous.setOnClickListener {
+            // To call the wifi setup page
+            roomNameSelected = null
+            wifiSetup()
+        }
+    }
+
+    // Final step of connectivity
+    override fun passwordLayoutSetup() {
+        password_layout.visibility = View.VISIBLE
+        room.visibility = View.GONE
+        room_wizard.background = context?.let { ContextCompat.getDrawable(it, R.drawable.border_blue) }
+        account_wizard.background = context?.let { ContextCompat.getDrawable(it, R.drawable.border_blue) }
```

Comment:
same at all places where context?.let is used, can be replaced by requireContext() for fragments